tests/Browser/BookingReviewTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Tests\Browser;
+
+use App\Models\BookingRequest;
+use App\Models\User;
+use App\Notifications\CommentNotification;
+use Database\Seeders\RolesAndPermissionsSeeder;
+use Database\Seeders\RoomSeeder;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Support\Facades\Mail;
+use Laravel\Dusk\Browser;
+use SlevomatCodingStandard\Helpers\Comment;
+use Tests\DuskTestCase;
+
+class BookingReviewTest extends DuskTestCase
+{
+    use DatabaseMigrations;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        (new RolesAndPermissionsSeeder())->run();
+        (new RoomSeeder())->run();
+        User::factory(1)->create()->first();
+        User::first()->assignRole('super-admin');
+    }
+
+    public function testUserCanViewBookingReviewPage()
+    {
+        $booking = BookingRequest::factory()
+            ->count(1)
+            ->hasReservations(random_int(1, 3))
+            ->create(["status" => BookingRequest::REVIEW])->first();
+
+        $this->browse(function (Browser $browser) use ($booking)
+        {
+            $browser->loginAs(User::first());
+            $browser->visit('/bookings/review')
+                ->clickLink('Open Details')
+                ->pause(5000)
+                ->assertPathIs('/bookings/' . $booking->id . '/review')
+                ->assertSee('Booking History');
+        });
+    }
+
+    public function testUserPostComments()
+    {
+        $booking = BookingRequest::factory()
+            ->count(1)
+            ->hasReservations(random_int(1, 3))
+            ->create(["status" => BookingRequest::REVIEW])->first();
+
+        $this->browse(function (Browser $browser) use ($booking)
+        {
+            $browser->loginAs(User::first());
+            $browser->visit('/bookings/review')
+                ->clickLink('Open Details')
+                ->pause(5000)
+                ->assertPathIs('/bookings/' . $booking->id . '/review')
+                ->assertSee('Booking History')
+                ->scrollTo('.ProseMirror')
+                ->click('.ProseMirror')
+                ->type('.ProseMirror', 'This is a test comment.')
+                ->pressAndWaitFor('@saveText')
+                ->releaseMouse()
+                ->scrollTo('.ProseMirror')
+                ->assertSee("This is a test comment.");
+        });
+
+        $this->assertDatabaseHas('email_log', [
+            'subject' => __('Comment Notification')
+        ]);
```

Comment:
No need to import a package. you can mock mail and notifcations : https://laravel.com/docs/8.x/mocking#notification-fake

```
class ExampleTest extends TestCase
{
    public function test_comment_send_notification()
    {
        Notification::fake();

        // Assert that no notifications were sent...
        Notification::assertNothingSent();
        
        // Perform dusk browsers stuff

        // Assert a notification was sent to the given users...
        Notification::assertSentTo(
            [$user], CommentNotification::class
        );

        // Assert a notification was not sent...
        Notification::assertNotSentTo(
            [$user], CommentNotification::class
        );
    }
}
```

Reply:
@a-drew that only works for feature & unit tests unfortunately.

The dusk test runner is a separate process from the app, so it has no access to intercept & mock calls

Reply:
ahhh I gotcha. ngl testing for notifications should've been a backend test, much faster. 
But with dusk you get a 2 for one kind of deal so 🤷

for any future facade related needs: https://github.com/NoelDeMartin/laravel-dusk-mocking